physioqc/interfaces/visualizations.py

```diff
@@ -0,0 +1,197 @@
+from typing import List, Literal, Tuple
+
+import matplotlib.pyplot as plt
+import numpy as np
+import peakdet as pk
+
+from ..metrics import multimodal
+
+
+def check_create_figure(
+    ax: plt.axes = None, figsize=(10, 5)
+) -> Tuple[plt.figure, plt.axes]:
+    """Helper function to check if a new figure should be created or to further
+    plot on axes.
+
+    Parameters
+    ----------
+    ax : plt.axes, optional
+        Check is axes object is present, if not creates new figure, by default None
+    figsize : tuple, optional
+        Size of the new figure (if created), by default (10, 5)
+
+    Returns
+    -------
+    Tuple[plt.figure, plt.axes]
+        Figure and axes objects of the plot.
+    """
+
+    if ax is None:
+        fig, ax = plt.subplots(1, 1, figsize=figsize)
+    else:
+        fig = None
+
+    return fig, ax
+
+
+def plot_raw_data(phys: pk.Physio, ax: plt.axes = None) -> Tuple[plt.figure, plt.axes]:
+    """Plots the raw data using peakdet.
+    Parameters
+    ----------
+    phys : pk.Physio
+        The data object, used for plotting.
+    ax : plt.axes, optional
+        axes object to use for plotting, by default None
+
+    Returns
+    -------
+    Tuple[plt.figure, plt.axes]
+        Figure and axes objects of the plot.
+    """
+    fig, ax = check_create_figure(ax, figsize=(7, 5))
+
+    ax = pk.plot_physio(phys, ax=ax)
+
+    return fig, ax
+
+
+def plot_average_peak(
+    phys: pk.Physio,
+    window: List = [-3, 3],
+    target_fs: float = 1000.0,
+    peak_dist: float = 1.0,
+    peak_thr: float = 0.1,
```

Comment:
In my opinion, a better way to implement this function would be to check if the Physio object already has undergone peak detection, and if so, to use the existing peaks. 

Right now, it seems like these peak detection variables are buried in a fairly unexpected place. It could improve functionality in the long run to add these as fields in the Physio object itself, since they seem to be frequently specified throughout this package and it is important to be consistent with them if they need to be altered for different data.

Reply:
I think we decided to re-do peak detection here, as it's done only for the plotting process.

Reply:
Did you test whether these parameters robustly detected peaks across tricky time-series and a number of physio-types? My concern is that the way they are specified in the function here may not translate to different data types (e.g. a relatively high heart rate, or data with an unexpected scaling or low SNR). Since these parameters will end up being embedded (maybe unexpectedly to the end user) inside the function call, they might not be obvious to change. If the user has spent some time optimizing peak detection with peak det, it makes sense to propagate those parameters forward here.

Reply:
Didn't really test this (don't really have a good grasp on the parameters). Just remembered, the reason for redoing peak detection was due to this bug https://github.com/physiopy/peakdet/issues/63, and higher fs than 1000Hz is too much for plotting. We've also been operating under the idea of applying QC to rawdata first, we did not want to pre-suppose a fixed pipeline.  
Until the bug is fixed, I'd suggest, to leave the function as is. But for the automated pipeline, I think we should have an autodetection of modality and set some default parameters accordingly, but also allow the parameters to be overridden by the user in the CLI.